langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/MetadataHandler.java

```diff
@@ -0,0 +1,93 @@
+package dev.langchain4j.store.embedding.pgvector;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.store.embedding.filter.Filter;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * Handle PV Store metadata.
+ */
+public interface MetadataHandler {
+
+    /**
+     * Definition used to create the metadata field(s) in embeddings table
+     *
+     * @return the sql clause that creates metadata field(s)
+     *         example:
+     *
+     */
+    String columnDefinition();
```

Comment:
```suggestion
    List<String> columnDefinitions();
```
Or `Map<String, ...>` like in `MetadataStorageConfig`